iphone/cli/commands/_build.js

```diff
@@ -4914,1233 +5033,1256 @@ iOSBuilder.prototype.writeDebugProfilePlists = function writeDebugProfilePlists(
-					const bootstrapJsonAbsolutePath = path.join(this.encryptJS ? this.buildAssetsDir : this.xcodeAppDir, bootstrapJsonRelativePath);
-					const bootstrapJsonString = JSON.stringify({ scripts: jsBootstrapFiles });
+			missingLaunchLogos.push({
+				description: desc,
+				file: dest,
+				width: spec.size,
+				height: spec.size,
+				required: false
+			});
 
-					if (this.encryptJS) {
-						this.jsFilesEncrypted.push(originalBootstrapJsonName); // original name
-						this.jsFilesToEncrypt.push(bootstrapJsonRelativePath); // encrypted name
-					}
+			if (!logged) {
+				logged = true;
+				this.logger.info(__n(
+					'Missing %s launch logo, generating missing launch logo from %%s',
+					'Missing %s launch logos, generating missing launch logos from %%s',
+					missingCount,
+					launchLogo ? 'LaunchLogo.png' : 'DefaultIcon.png'
+				));
+			}
 
-					if (!fs.existsSync(bootstrapJsonAbsolutePath) || (bootstrapJsonString !== fs.readFileSync(bootstrapJsonAbsolutePath).toString())) {
-						this.logger.debug(__('Writing %s', bootstrapJsonAbsolutePath.cyan));
+			if (launchLogo) {
+				this.logger.info('  ' + desc);
+			}
+		});
+	}
 
-						fs.ensureDirSync(path.dirname(bootstrapJsonAbsolutePath));
-						fs.writeFileSync(bootstrapJsonAbsolutePath, bootstrapJsonString);
-					} else {
-						this.logger.trace(__('No change, skipping %s', bootstrapJsonAbsolutePath.cyan));
-					}
+	await this.writeAssetContentsFile(path.join(assetCatalogDir, 'Contents.json'), {
+		images: images,
+		info: {
+			version: 1,
+			author: 'xcode'
+		}
+	});
 
-					this.unmarkBuildDirFile(bootstrapJsonAbsolutePath);
-				},
-			], next);
-		},
+	// If there are no missing launch logos, we're done
+	if (!missingLaunchLogos.length) {
+		return [];
+	}
 
-		function writeAppProps() {
-			this.logger.info(__('Writing app properties'));
+	if (!this.forceRebuild) {
+		this.logger.info(__('Forcing rebuild: launch logos changed since last build'));
+		this.forceRebuild = true;
+	}
+
+	if (!this.buildOnly && (this.target === 'device' || this.target === 'simulator')) {
+		this.logger.warn(__('If this app has been previously installed on this %s, you may need restart it to see the latest launch logo', this.target));
+		this.logger.warn(__('iOS renders and caches the launch screen to a PNG image that seems to only be invalidated by restarting iOS'));
+	}
 
-			const appPropsFile = this.encryptJS ? path.join(this.buildAssetsDir, '_app_props__json') : path.join(this.xcodeAppDir, '_app_props_.json');
-			const props = {};
+	if (!launchLogo) {
+		// just use the DefaultIcon.png to generate the missing LaunchLogos
+		return missingLaunchLogos;
+	}
 
-			if (this.encryptJS) {
-				this.jsFilesEncrypted.push('_app_props_.json'); // original name
-				this.jsFilesToEncrypt.push('_app_props__json'); // encrypted name
+	// Resize the launch logo to the other dimensions needed
+	await new Promise((resolve, reject) => {
```

Comment:
<!--
  0 failure: 
  1 warning:  iphone/cli/comman...
  
  
  DangerID: danger-id-Danger;
  File: iphone/cli/commands/_build.js;
  Line: 6114;
-->

- :warning: iphone/cli/commands/_build.js line 6114 – 'reject' is defined but never used. Allowed unused args must match /^_.+/u. (no-unused-vars)